src/scipp/experimental/plotting/wrappers.py

```diff
@@ -26,14 +27,19 @@ def _to_data_array(obj):
 
 def _convert_if_not_none(x, unit):
     if x is not None:
-        return to_unit(x, unit=unit)
+        return to_unit(number_to_variable(x), unit=unit)
```

Comment:
No here and now, but we should consider adding the equivalent to `np.asarray` to scipp.